keps/sig-node/5075-dra-dynamic-provisioning-support/README.md

```diff
@@ -0,0 +1,1055 @@
+
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+A motivating use case for supporting dynamically-provisioned device is a virtual network device when to-be-allocated device will be created or partitioned on demand (on claim). The orginal discussion is in [this PR's comment thread](https://github.com/kubernetes-sigs/cni-dra-driver/pull/1#discussion_r1889265214) and the limitation of current implementation has been addressed [here](https://github.com/kubernetes-sigs/cni-dra-driver/pull/1#discussion_r1890166449). The virtual network device is created and configured once the CNI is called based on the information of the master network device. The configured information specific to the generated device should be at the ResourceClaim not the ResourceSlice. However, the device in the ResourceCliam is now present only the device listed in the ResourceSlice and there is no attribute to differentiate between the master device and the actual provisioned device.
+
+There are also similar use cases for requesting the device provision based on a specific root or device type in other contexts including [Peer Pods](https://github.com/confidential-containers/cloud-api-adaptor/blob/main/docs/architecture.md) and [Composable Disaggregated Infrastructure (CDI)](https://www.wwt.com/article/an-introduction-to-composable-disaggregated-infrastructure). 
+
+This feature is also beneficial for the other provisonable devices those are not with a scope of [KEP-4815](https://github.com/kubernetes/enhancements/issues/4815). For instance, this feature will be allow reserving memory fraction of virtual GPU in [the AWS virtual GPU device plugin](https://github.com/awslabs/aws-virtual-gpu-device-plugin) via DRA. The number of provision can be limited if needed.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Introduce an ability to allocating on-demand provisioned devices via DRA. This should cover the use case of macvlan, ipvlan in DRA driver for CNI and virtual accelerator devices with on-demand memory fraction. 
```

Comment:
Suggestions for grammar/clarity:  Introduce an ability to allocating on-demand provisioned devices via DRA. This should cover the use cases of macvlan or ipvlan in a DRA driver for CNI and virtual accelerator devices with on-demand memory fraction.